### DISCUSSION TITLE
Understanding Behavior Differences in Alias Types

### DISCUSSION THREAD

**Discussion by sam-mosleh** [pydantic/pydantic#7941] [Question]
I want to seek some clarification on the behavior differences between two distinct types of aliases in the new version of the library. I must say, I find the introduction of these different types of aliases a brilliant design choice.
In my experience, I've noticed that these differences can cause unexpected issues for users who are using a `TypeAdapter` or `BaseModel` when working with JSON. To show you what I mean, I've included a code example below:

```python
import pydantic

class AliasModel(pydantic.BaseModel):
    a: int = pydantic.Field(alias="b")

class SerializationAliasModel(pydantic.BaseModel):
    a: int = pydantic.Field(serialization_alias="b")

alias_model = AliasModel(b=10)
serialization_alias_model = SerializationAliasModel(a=10)

assert (
    AliasModel.model_validate_json(alias_model.model_dump_json(by_alias=True))
    == alias_model
)

SerializationAliasModel.model_validate_json(
    serialization_alias_model.model_dump_json(by_alias=True)
)  # This raises a ValidationError

```
Could you please provide guidance on how one can effectively utilize `serialization_alias` for dumping and validation, without relying on multiple `AliasChoices`?

**Comment by AlessandroMiola** [comment]
The `serialization_alias` is meant to **override the `alias` specification when _serializing_ data `by_alias`**; otoh, differently from the _plain_ `alias`, you can't use it for _validation_/_deserialization_ (which is why your second case fails).

- With _plain_ `alias`, you would:
    - _instantiate_ model fields by `alias` 
    - _validate_ data by `alias`
    - _serialize_ data by `field_name` (when `by_alias=False`) or possibly by `alias` (when `by_alias=True`)

- With `serialization_alias`, you would:
    - _instantiate_ model fields by `field_name` 
    - _validate_ data by `field_name`
    - _serialize_ data by `field_name` (when `by_alias=False`) or possibly by `serialization_alias` (when `by_alias=True`)

- With both `alias` and `serialization_alias`, you would:
    - _instantiate_ model fields by _plain_ `alias` 
    - _validate_ data by _plain_ `alias` 
    - _serialize_ data by `field_name` (when `by_alias=False`) or possibly by `serialization_alias` (when `by_alias=True`), **thus overriding the `alias` specification**.

For further references, please find the related documentation at https://docs.pydantic.dev/latest/concepts/fields/#field-aliases.
Also, https://github.com/pydantic/pydantic/discussions/9509#discussioncomment-9618274 might be possibly useful in discerning among the possible behaviours.
